src/escrow_contract/src/types/account_note.nr

```diff
@@ -0,0 +1,54 @@
+use dep::aztec::{
+    context::PrivateContext,
+    macros::notes::note,
+    note::{
+        note_header::NoteHeader, note_interface::NullifiableNote,
+        utils::compute_note_hash_for_nullify,
+    },
+    oracle::random::random,
+    prelude::AztecAddress,
+    protocol_types::{
+        constants::GENERATOR_INDEX__NOTE_NULLIFIER,
+        hash::poseidon2_hash_with_separator,
+        traits::{Empty, Packable, Serialize, ToField},
+    },
+};
+
+#[note]
+#[derive(Serialize)]
+pub struct AccountNote {
+    owner: AztecAddress,
+    secret: Field,
+    randomness: Field,
+}
+
+impl NullifiableNote for AccountNote {
+    fn compute_nullifier(self, _: &mut PrivateContext, note_hash_for_nullify: Field) -> Field {
+        poseidon2_hash_with_separator(
+            [note_hash_for_nullify, self.owner.to_field()],
+            GENERATOR_INDEX__NOTE_NULLIFIER as Field,
+        )
+    }
+
+    unconstrained fn compute_nullifier_without_context(self) -> Field {
+        let note_hash_for_nullify = compute_note_hash_for_nullify(self);
+        poseidon2_hash_with_separator(
+            [note_hash_for_nullify, self.owner.to_field()],
+            GENERATOR_INDEX__NOTE_NULLIFIER as Field,
+        )
+    }
+}
+
+impl Eq for AccountNote {
+    fn eq(self: Self, other: Self) -> bool {
+        (self.owner.to_field() == other.owner.to_field())
+    }
+}
```

Comment:
i'm not sure this is needed at all, `Eq` isn't being used to "filter notes" so we can remove it, and `NullifiableNote` could be not needed if it's PrivateImmutable, tho i'm not fully sure about this last, we can think what's the "nullifying" design, tho v77 comes already having implemented this iirc.